doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,571 @@
+that which is different from the initially generated dynamic profile.
+
+When the generator is run, and determines that a new profile has been added,
+we'll need to make sure to add the profile to the user's settings file. This
+will create an easy point for users to customize the dynamic profiles. When
+added to the user settings, all that will be added is the `name`, `guid`, and
+`source`.
+
+Additionally, a user might not want a dynamic profile generator to always run.
+They might want to keep their Azure connections visible in the list of profiles,
+even if its no longer a valid target. Or they might want to not automatically
+connect to Azure to find new instances every time they launch the terminal. To
+enable scenarios like this, we'll add an additional setting,
+`disabledProfileSources`. This is an array of guids. If any guids are in that
+list, then those dynamic profile generators _won't_ be run, supressing those
+profiles from appearing in the profiles list.
+
+#### What if a dynamic profile is removed, but it's the default?
+
+We should probably add an extra validation step at the end of serializing to
+ensure that the default profile isn't a hidden profile. If the user's set
+default profile is a hidden profile, we'll just set the first non-hidden profile
+as the default profile.
+
+### Powershell Core & the Defaults
+
+How do we handle the potential existence of powershell core in this model?
+Powershell core is unique as far as the default profiles goes - it may or may
+not exist on the user's system. Not only that, but depending on the user's
+install of powershell core, it might have a path in either `Program Files` or
+`Program Files(x86)`.
+
+Additionally, if it _is_ installed, we set it as the default profile instead of
+Windows Powershell.
+
+Should we include it in the default profiles? If we wanted to do this, we'd need
+to:
+* hide it in the user settings if it's not installed.
+* Set the commandline in the user settings. If the user deletes this value, then
+  that profile would be unable to launch, unlike all the other profiles that are
+  in the user settings (by default).
+* Make sure to set the default profile in the user settings.
+
+Or, should we only add it to the user settings?
+* We'd only generate it in the user settings if it's installed.
+* Set the commandline in the user settings. If the user deletes this value, then
+  that profile would be unable to launch, unlike all the other profiles that are
+  in the user settings (by default).
+* Make sure to set the default profile in the user settings.
+
+Currently there are only a few things that we actually set for the Powershell
+Core profile - the name, commandline, icon, and the color scheme. I don't
+believe it would add too much bloat to the user's settings to generate it in the
+user's settings by default.
+
+As a third option, we could treat the Powershell Core profile like a dynamic

```

Comment:
As I was reading the above, I was hoping you'd just say "treat it like another dynamic provider." Do this.

Reply:
There's a lot of consensus on that. I;ll remove the other options.